### ISSUE TITLE
Task 3.3: STDP Training for Survival SNN

### ISSUE THREAD

**Issue by nfriacowboy** [nfriacowboy/ethical-snn-research#9]
**Priority**: 🟡 Important  
**Status**: ⏸️ Not started  
**Estimated time**: 6 hours  
**Week**: 3 - Organism Base & Survival SNN

## Subtasks
- [ ] Create `src/training/stdp_trainer.py`
  - [ ] Class `STDPTrainer`
  - [ ] Method: `update_weights(pre_spikes, post_spikes, reward)`
  - [ ] Implement reward-modulated STDP
- [ ] Integration with simulation loop
- [ ] Add unit tests: `tests/test_stdp_trainer.py`

## Acceptance Criteria
```python
trainer = STDPTrainer(snn, config)
pre_spikes = torch.randn(30)
post_spikes = torch.randn(5)
trainer.update_weights(pre_spikes, post_spikes, reward=10)
# Weights should change based on spike timing
```

**Dependencies**: Task 3.2  
**Blocks**: Task 5.1
